src/bodyElement.ts

```diff
@@ -278,6 +289,31 @@ const parse = (context: Context, atoms?: Atoms) =>
             return new Ok({ kind: ElementKind.Embed, html: embedHtml, alt: fromNullable(alt) });
         }
 
+        case ElementType.MEMBERSHIP: {
+            const {
+                linkText,
+                originalUrl: url,
+                price,
+                start,
+                image
+            } = element.membershipTypeData ?? {};
+
+            if (!linkText || !url) {
+                return new Err('No linkText or originalUrl field on membershipTypeData');
+            }
+
+            const formattedDate = start ? formatDate(new Date(start?.iso8601)) : undefined;
```

Comment:
I think this may sidestep the possibility of an invalid date instance, either from `start` being `undefined` or if the `iso8601` field is mangled somehow:

```suggestion
            const date = new Date(start?.iso8601);
            const formattedDate = !isNaN(date) ? formatDate(date) : undefined;
```